fsi/fsi.go

```diff
@@ -210,6 +211,11 @@ func (fsi *FSI) Unlink(dirPath, refStr string) error {
 	}
 
 	defer func() {
+		// clean up low value files before running Remove
+		err := removeLowValueFiles(dirPath)
+		if err != nil {
+			log.Errorf("removing directory: %s", err.Error())
+		}
```

Comment:
I think this would be better if it lived here: https://github.com/qri-io/qri/blob/master/lib/datasets.go#L793. For one, this behavior isn't directly needed by `fsi.Unlink`, only by `lib.Remove`. If a user runs `qri fsi unlink` then this function also gets called, via the `Unlink` function in `lib/fsi.go`, but that shouldn't be deleting files. Furthermore, having this functionality here means that it doesn't respect the `--keep-files` flag, which if true shouldn't modify any files at all.

Would also be great to have a test for this in `cmd/`, which creates a directory, runs `qri init`, runs `save`, writes a `.DS_Store` file to that directory, then runs `remove --all` to make sure the whole thing is deleted. Copying `TestRemoveOneRevisionFromWorkingDirectory` then modifying it would be a great start.

Reply:
Agreed on moving the call site to https://github.com/qri-io/qri/blob/master/lib/datasets.go#L793, but I think the logic of what we mean by `RemoveAll` should live in the `fsi` package. To me the best of both worlds is if `fsi` had an exported function:

```go
package fsi

// RemoveAll is like os.RemoveAll, but it also ignores "low-value files"
func RemoveAll(path string) error {
  // ...
}
```

Then call that function from lib.

Reply:
I took a bit of a split approach with `datastore` retaining control but moving the execution logic to `fsi`.